### PR TITLE
update status of spiaccesscheck when token lookup fails

### PR DESCRIPTION
**Pull request by sparkoo**
### What does this PR do?
Reports error to spiaccesscheck status when token lookup fails and don't reconcile CR again.

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-161

### How to test this PR?

 1. manually upload the token with test/test credentials for quay.io token
```
curl -v \
  -H 'Content-Type: application/json' \
  -H "Authorization: bearer "$K8S_TOKEN \
  -d "{ \"access_token\": \"test\",\"username\": \"test\"}" \
  $OAUTH_URL'/token/'$SPI_NAMESPACE'/'$SPI_TOKEN_NAME
```
 2. create spiaccesscheck for quay.io repository
 3. you should see error in spiaccesscheck status that lookup failed. If repository is public, it should still be detected.

### CODE REVIEW

File: pkg/serviceprovider/quay/quay_test.go [hunk 1]
```diff
@@ -505,13 +505,13 @@ func TestCheckRepositoryAccess(t *testing.T) {
...
+		assert.NoError(t, err)
 		assert.NotNil(t, status)
-		assert.False(t, status.Accessible)
+		assert.True(t, status.Accessible)
 		assert.Equal(t, api.SPIRepoTypeContainerRegistry, status.Type)
 		assert.Equal(t, api.ServiceProviderTypeQuay, status.ServiceProvider)
-		assert.Equal(t, api.SPIAccessCheckAccessibilityUnknown, status.Accessibility)
-		assert.Equal(t, api.SPIAccessCheckErrorUnknownError, status.ErrorReason)
+		assert.Equal(t, api.SPIAccessCheckAccessibilityPublic, status.Accessibility)
+		assert.Equal(t, api.SPIAccessCheckErrorTokenLookupFailed, status.ErrorReason)
```
Comment (metlos) [inline pkg/serviceprovider/quay/quay_test.go:514]: I'm not sure how I would parse this as a user... Was there an error and "public" was left there as the default option or was the access determined to be public and the error was left there as an "artifact" of processing?

In either case I'd be confused whether this is a valid state...

Comment (sparkoo) [reply pkg/serviceprovider/quay/quay_test.go]: fixed

File: pkg/serviceprovider/github/github.go [hunk 2]
```diff
@@ -182,7 +182,11 @@ func (g *Github) CheckRepositoryAccess(ctx context.Context, cl client.Client, ac
 	tokens, lookupErr := g.lookup.Lookup(ctx, cl, accessCheck)
 	if lookupErr != nil {
 		lg.Error(lookupErr, "failed to lookup token for accesscheck", "accessCheck", accessCheck)
-		return status, fmt.Errorf("failed to lookip token for accesscheck %s/%s: %w", accessCheck.Namespace, accessCheck.Name, lookupErr)
+		if !publicRepo {
```
Comment (metlos) [inline pkg/serviceprovider/github/github.go:185]: Btw why do we even continue here if we already determined the repository was public? Do we learn any new information by doing the token lookup?

File: pkg/serviceprovider/quay/quay.go [hunk 3]
```diff
@@ -240,8 +243,10 @@ func (q *Quay) CheckRepositoryAccess(ctx context.Context, cl client.Client, acce
 		case http.StatusUnauthorized, http.StatusForbidden:
 			lg.Info("quay.io request unauthorized. Probably private repository for we don't have a token.")
```
Comment (metlos) [inline pkg/serviceprovider/quay/quay.go:244]: "Probably a private repository for which we don't have a token."

**Review by metlos — COMMENTED** [review]
LGTM apart from the two nitpicks..

File: pkg/serviceprovider/quay/quay.go (hunk 3 above)
Comment (sparkoo) [reply pkg/serviceprovider/quay/quay.go]: fixed

File: pkg/serviceprovider/github/github.go (hunk 2 above)
Comment (sparkoo) [reply pkg/serviceprovider/github/github.go:186]: idea was that we can find more information. But it's not actually needed. I've updated the code so it returns immediately at `if publicRepo`

**Review by metlos — APPROVED** [review]

**Pull request merged**